test/Test/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -254,42 +255,65 @@ prop_entries_all page@(PageLogical xs) bs =
 
 prop_big_insert :: Key -> Maybe BlobRef -> Property
 prop_big_insert k blobref =
-    rawPageLookup rawpage k' === case blobref of
-        Nothing            -> LookupEntryOverflow (Entry.Insert (SerialisedValue $ RB.fromByteString (BS.take size v))) (fromIntegral sfxSize)
-        Just (BlobRef x y) -> LookupEntryOverflow (Entry.InsertWithBlob (SerialisedValue $ RB.fromByteString (BS.take size v)) (BlobSpan x y)) (fromIntegral sfxSize)
+    case rawPageLookup rawpage k' of
+      LookupEntryOverflow (Entry.Insert (SerialisedValue v')) suffixlen
+        | Nothing <- blobref
+        -> v === RB.toByteString v' <>
+                 BS.take (fromIntegral suffixlen) overflowPagesBS
+
+      LookupEntryOverflow (Entry.InsertWithBlob (SerialisedValue v')
+                                                (BlobSpan x' y'))
+                          suffixlen
+        | Just (BlobRef x y) <- blobref
+        -> v === RB.toByteString v' <>
```

Comment:
is this something a generalised `compareOpEntry` would do?

Reply:
I've made this use clearer utils:
```
-- | Check that an 'Operation' and optional 'Blobref' (from the reference
-- implementation) is equivalent to (from the real implementation) the
-- combination of an 'Entry', containing the prefix of a value, plus a given
-- length suffix from the overflow pages.
equivalentOpEntry :: Operation -> Maybe BlobRef
                  -> Entry SerialisedValue BlobSpan
                  -> BS.ByteString
                  -> Int
                  -> Property

-- | Check that a value is equality to the combination of a 'SerialisedValue'
-- prefix, plus a given length of suffix from the overflow pages.
--
equivalentValue :: BS.ByteString -> Value -> SerialisedValue -> Int -> Property

equivalentBlobRefSpan :: BlobRef -> BlobSpan -> Property
```